components/net/fetch/methods.rs

```diff
@@ -872,3 +879,44 @@ fn is_bad_port(port: u16) -> bool {
 
     BAD_PORTS.binary_search(&port).is_ok()
 }
+
+/// <https://w3c.github.io/webappsec-upgrade-insecure-requests/#upgrade-request>
+fn should_upgrade_request_to_potentially_trustworty(
+    request: &mut Request,
+    context: &FetchContext,
+) -> bool {
+    if request.is_navigation_request() {
+        if !request.current_url().is_origin_trustworthy() ||
+            !context
+                .state
+                .hsts_list
```

Comment:
Can you add spec comments to this function? Like this:
```rust
// Step 1. If request is a navigation request, append a header named Upgrade-Insecure-Requests with a value of 1 to request’s header list if any of the following criteria are met:
// * request’s URL is not a potentially trustworthy URL
// * request’s URL's host is not a preloadable HSTS host
```

Also please either implement the HSTS check or add a TODO about it (same for other steps from the spec that this is missing)